community/assessments/projects/k8gb/self-assessment.md

```diff
@@ -0,0 +1,234 @@
+
+### Actors
+
+The individual parts of k8gb that interact to provide the desired functionality.
+
+- CoreDNS
+  - Role: Embedded [custom CoreDNS](https://github.com/k8gb-io/coredns-crd-plugin) to serve DNS requests.
+  - Isolation: runs as its own Pod
+- ExternalDNS
+  - Role: Integrated ExternalDNS to automate zone delegation configuration.  
+  - Isolation: runs as its own Pod
+- k8gb Controller
+  - Role: Coordinates logic according to the GSLB strategy.
+  - Isolation: runs as its own Pod
+
+### Actions
+
+The steps that k8gb performs in order to provide the desired functionality. See https://www.k8gb.io/docs/images/k8gb-components.svg. 
+
+- CoreDNS
+  - Functionality:
+    - Serves external DNS requests with dynamically constructed DNS responses.
+- ExternalDNS
+  - Functionality:
+    - Reads data from DNSEndpoint CR to update an external DNS provider (e.g., Route53).
+- k8gb Controller
+  - Functionality:
+    - Watches all namespaces for `Gslb` custom resources.
+    - Creates DNSEndpoint CR populated with information from `Gslb` Ingress status (application FQDN and active IP addresses, which are used for dynamic A record composition).
+    - Creates DNSEndpoint to configure DNS zone delegation in an external DNS provider.
+
+### Goals
+
+The intended goals of k8gb, including the security guarantees it provides.
+
+#### Secure and Verified Builds
+
+Ensure all K8GB releases are signed and verified to guarantee authenticity and integrity, protecting users from tampered or malicious builds.
+
+#### Minimal Attack Surface
+
+Expose only the necessary ports for GSLB operations, such as DNS (53/tcp and 53/udp), to reduce the attack surface and enhance security.
+
+#### Secure Deployment Practices
+
+Provide secure default configurations and documentation to help users deploy K8GB in a way that aligns with Kubernetes security best practices.
+
+These goals aim to make K8GB a reliable and secure solution for global load balancing while minimizing risks and ensuring trust in the project's artifacts.
+
+See also [Intended use](#intended-use).
+
+### Non-goals
+
+Non-goals that a reasonable reader of k8gb’s documentation could believe may be in scope.
+The k8gb project is focused on providing global load balancing and failover capabilities for Kubernetes applications. However, there are certain security-related features and responsibilities that are intentionally out of scope. Below are the non-goals, along with explanations for why they are not within the project's scope:
+
+#### Traffic Handling and TLS Termination:
+
+k8gb does not pass application traffic through itself or handle TLS/HTTP connections directly. These responsibilities are offloaded to referenced networking resources, such as Kubernetes Ingress controllers or service meshes.
```

Comment:
So, it directs requests for a DNS name.  This certainly (indirectly) impacts the sorts of issues you describe, right?